### ISSUE TITLE
Refactoring crashes when typing `[]` in class body

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#37584]
```CS
public class Class1
{
   // Uncomment 
   // []
}

public class TestAttribute : Attribute
{
}
```

Uncommenting that line crashes a whole bunch of IDE refactorings. This is one such stack trace:
```
System.ArgumentOutOfRangeException : 'end' must not be less than 'start'
Parameter name: end
   at Microsoft.CodeAnalysis.Text.TextSpan.FromBounds(Int32 start,Int32 end)
   at Microsoft.CodeAnalysis.CodeRefactorings.AbstractRefactoringHelpersService.GetSpanWithoutAttributes(SyntaxNode node,SyntaxNode root,ISyntaxFactsService syntaxFacts)
   at async Microsoft.CodeAnalysis.CodeRefactorings.AbstractRefactoringHelpersService.TryGetSelectedNodeAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.AbstractRefactoringHelpersService.TryGetSelectedNodeAsync[TSyntaxNode](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PullMemberUp.CSharpPullMemberUpCodeRefactoringProvider.GetSelectedNodeAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.AbstractPullMemberUpRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

Int Preview 16.3 Preview 2.0 29128.8

**Comment by petrroll** [comment]
I'll double check & add tests but it's most likely been fixed here: https://github.com/dotnet/roslyn/pull/37295/commits/696a64a996dbec48b4e9a897acbebca561c788bd

Do we need to backport?

**Comment by petrroll** [comment]
Fixed in #37600 (backport with tests) and #37295 (master, with tests in #37588)
